apps/web/vibes/soul/primitives/calendar.tsx

```diff
@@ -0,0 +1,49 @@
+import { ComponentPropsWithoutRef } from 'react'
+import { DayPicker } from 'react-day-picker'
+
+import clsx from 'clsx'
+import { ChevronLeftIcon } from 'lucide-react'
+
+export function Calendar({
+  className,
+  classNames,
+  ...props
+}: ComponentPropsWithoutRef<typeof DayPicker>) {
+  return (
+    <DayPicker
+      className={clsx(
+        'box-content w-[280px] rounded-lg border border-contrast-100 bg-background p-3',
+        className
+      )}
+      classNames={{
+        months: 'relative',
+        month_caption: 'flex justify-center w-full font-medium pb-0.5',
+        nav: 'absolute flex justify-between w-full',
+        button_next:
+          'rotate-180 rounded-full focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-foreground',
+        button_previous:
+          'rounded-full focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-foreground',
+        month_grid: 'flex flex-col gap-0.5',
+        weeks: 'flex flex-col gap-0.5',
+        weekdays: 'flex',
+        weekday: 'flex h-10 w-10 items-center justify-center text-xs font-medium',
+        week: 'flex',
+        day: 'h-10 w-10 flex text-xs font-medium group p-0',
+        day_button:
+          'h-full w-full flex items-center justify-center rounded-full group-data-[selected=true]:bg-primary group-data-[selected=true]/middle:bg-transparent hover:border hover:border-contrast-200 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-foreground disabled:hover:border-none',
+        disabled: 'text-contrast-300',
+        outside: 'text-contrast-300',
+        range_start: 'bg-gradient-to-l from-primary-highlight',
+        range_middle: 'bg-primary-highlight group/middle',
+        range_end: 'bg-gradient-to-r from-primary-highlight',
+        ...classNames,
+      }}
+      components={{
+        Chevron: () => <ChevronLeftIcon className="h-5 w-5" strokeWidth={1} />,
```

Comment:
Is it a concern that on every render the `Chevron` component will have a different identity?

Reply:
Seems to be the [way to implement it](https://daypicker.dev/guides/custom-components#implementing-a-custom-component). What would you suggest?

Reply:
If this is how they document it, then I have no issues here. They might be calling it as a "render function" instead of using it as a component (I hope...)

Reply:
Ugh, looks like they don't: https://github.com/gpbl/react-day-picker/blob/903745cb6c1332818ef558729c0c598a18563767/src/components/Nav.tsx#L44

The solution here would be to define the component outside of `Calendar` so that it's not a new instance on each render. I'm surprised this library has this React antipattern in their docs like this.

Reply:
I guess it's not a big deal if the chevron remounts but it's just bad practice

Reply:
Here's somebody running into the problem: https://github.com/gpbl/react-day-picker/issues/839#issuecomment-444150953

I say let's just pull out the `components` object outside the component as a top level `const`